Dockerfile

```
@@ -105,5 +105,5 @@ ENV IN_DOCKER=1
 # also works for GitLab CI/CD
 # https://gitlab.com/gitlab-org/gitlab-runner/-/blob/4c42e96/shells/bash.go#L18-37
 # https://gitlab.com/gitlab-org/gitlab-runner/-/blob/4c42e96/shells/bash.go#L288
-ENTRYPOINT ["/bin/bash", "-c", "which -- \"$0\" &>/dev/null && exec \"$0\" \"$@\" || exec cml \"$0\" \"$@\""]
+ENTRYPOINT ["/bin/bash", "-c", "echo \"$0\" | grep -qE '^(pr|publish|runner|send-(comment|github-check)|tensorboard-dev)$' && exec cml \"$0\" \"$@\" || exec \"$0\" \"$@\""]
```

```suggestion
ENTRYPOINT ["/bin/bash", "-c", "echo \"$0\" | grep -qE '^(pr|publish|runner|send-(comment|github-check)|tensorboard-dev|--?(-|\w)+(=.+)?)$' && exec cml \"$0\" \"$@\" || exec \"$0\" \"$@\""]
```

Perhaps also when `$0` is an option?

---

nice idea

```suggestion
ENTRYPOINT ["/bin/bash", "-c", "echo \"$0\" | grep -qE '^(pr|publish|runner|send-(comment|github-check)|tensorboard-dev|--?[[:alnum:]].*)$' && exec cml \"$0\" \"$@\" || exec \"$0\" \"$@\""]
```

---

Please note that [`[:alnum:]`](https://www.gnu.org/software/grep/manual/html_node/Character-Classes-and-Bracket-Expressions.html#index-alnum-character-class) doesn't include hyphens like in `--long-option`

---

Also, [`\w`](https://www.gnu.org/software/grep/manual/html_node/The-Backslash-Character-and-Special-Expressions.html) should be an alias for `[_[:alnum:]]` and still don't include hyphens.

---

techincally `--?[[_:alpha:]].*` is most correct but `--?\w.*` is neater.

---

It still has some flaws, but it's definitely neater. Let's let `yargs` do teh work for us and print a nice error.

`echo "--_验证错误" | grep -E '^--?\w.*$'`,

---

yar :black_flag: